lib/ansible/plugins/terminal/ios.py

```diff
@@ -58,7 +58,10 @@ def on_open_shell(self):
             raise AnsibleConnectionFailure('unable to set terminal parameters')
 
     def on_become(self, passwd=None):
-        if self._get_prompt().endswith(b'#'):
+        if self._get_prompt():
```

Comment:
Better save it into a var rather then re-running function twice.

Reply:
thanks n updated the change.